tokio/src/task/local.rs

```diff
@@ -1160,9 +1160,15 @@ impl LocalState {
 
     #[track_caller]
     fn assert_called_from_owner_thread(&self) {
-        // FreeBSD has some weirdness around thread-local destruction.
+        // BSDs, illumos, and Solaris have some weirdness around thread-local
+        // destruction.
         // TODO: remove this hack when thread id is cleaned up
-        #[cfg(not(any(target_os = "openbsd", target_os = "freebsd")))]
+        #[cfg(not(any(
```

Comment:
Hmm, the actual code looks like this:
```
            context::thread_id()
                .map(|id| id == self.owner)
                .unwrap_or(true),
```
here it's using `map` to handle the case where getting the thread local fails. Not sure why these platforms fail in some other way. Anyway, since we already need this for other platforms, it doesn't seem like an issue to add more platforms here.

Reply:
Yeah, if memory serves, the problem is actually that on BSDs (and apparently, illumos/Solaris), getting the thread local apparently *doesn't* fail as we would expect, and it instead gets a new value or something. I vaguely recall @carllerche trying to figure this out while working on #5179, and I think we couldn't really figure out what was going on there and kinda gave up...